core/src/main/java/nl/nn/adapterframework/stream/Message.java

```diff
@@ -466,12 +459,57 @@ public synchronized byte[] getMagic(int readLimit) throws IOException {
 	}
 
 	private byte[] readBytesFromCharacterData(int readLimit) throws IOException {
-		String characterData = asString();
-		if (characterData.isEmpty()) {
+		if (request instanceof Reader) {
+			if (!((Reader) request).markSupported() || request instanceof TeeReader) {
+				request = new BufferedReader((Reader)request, readLimit);
+			}
+			Reader reader = (Reader) request;
+			reader.mark(readLimit);
+			try {
+				return readBytesFromReader(readLimit);
```

Comment:
Dit is niet zo handig, je kan/moet hier de reader meegeven aan de `readBytesFromReader` anders gebruikt hij niet de juiste markSupportedReader.